### DISCUSSION TITLE
How do i completely uninstall bun?

### DISCUSSION THREAD

**Discussion by nooblyf** [oven-sh/bun#4956] [Q&A] [answered]
I messed up my local cache so i'm thinking of doing a fresh install, does anyone know how do i do that?

**Comment by thansk** [comment]
Cache is in `~/.bun/install/cache`, you can try deleting that.

**Comment by allmtz** [comment, marked as answer]
From the docs, to completely uninstall for macOS, Linux and WSL:
```
$ rm -rf ~/.bun 
```
Then just reinstall
```
$ curl -fsSL https://bun.sh/install | bash 
```

**Comment by babakfp** [comment]
## Windows

I uninstalled Bun by dragging the `C:\Users\<username>\.bun\uninstall.ps1` file into Powershell (v7) and running it (press `Enter`).

If you tried uninstalling it by manually removing the `.bun` folder, reinstall it and do it the above way because https://github.com/oven-sh/bun/discussions/9802#discussion-6447834.

**Comment by Minecatr** [comment]
also if you installed bun with snap run `snap remove bun` you might need sudo idk.
